src/polyswarm_api/resources.py

```diff
@@ -0,0 +1,668 @@
+            api,
+            parameters,
+            result_parser=cls,
+        ).execute()
+
+    @classmethod
+    def rescan(cls, api, hash_value, hash_type, scan_config=None):
+        parameters = {
+            'method': 'POST',
+            'url': '{}/consumer/submission/{}/rescan/{}/{}'.format(api.uri, api.community, hash_type, hash_value),
+        }
+        if scan_config:
+            parameters.setdefault('data', {})['scan-config'] = scan_config
+        return core.PolyswarmRequest(
+            api,
+            parameters,
+            result_parser=cls,
+        ).execute()
+
+    @classmethod
+    def rescanid(cls, api, submission_id, scan_config=None):
+        parameters = {
+            'method': 'POST',
+            'url': '{}/consumer/submission/{}/rescan/{}'.format(api.uri, api.community, int(submission_id)),
+        }
+        if scan_config:
+            parameters.setdefault('data', {})['scan-config'] = scan_config
+        return core.PolyswarmRequest(
+            api,
+            parameters,
+            result_parser=cls,
+        ).execute()
+
+    @classmethod
+    def lookup_uuid(cls, api, submission_id):
+        return core.PolyswarmRequest(
+            api,
+            {
+                'method': 'GET',
+                'url': '{}/consumer/submission/{}/{}'.format(api.uri, api.community, int(submission_id)),
+            },
+            result_parser=cls,
+        ).execute()
+
+    @classmethod
+    def metadata_rerun(cls, api, hashes, analyses=None, skip_es=None):
+        parameters = {
+            'method': 'POST',
+            'url': '{}/consumer/metadata'.format(api.uri),
+            'json': {'hashes': hashes},
+        }
+        if analyses:
+            parameters['json']['analyses'] = analyses
+        if skip_es:
+            parameters['json']['skip_es'] = skip_es
+        return core.PolyswarmRequest(
+            api,
+            parameters,
+            result_parser=cls,
+        ).execute()
+
+    def __str__(self):
+        return "ArtifactInstance-<%s>" % self.hash
+
+    @property
+    def malicious_assertions(self):
+        if not self._malicious_assertions:
+            self._malicious_assertions = [a for a in self.assertions if a.mask and a.verdict]
+        return self._malicious_assertions
+
+    @property
+    def benign_assertions(self):
+        if not self._benign_assertions:
+            self._benign_assertions = [a for a in self.assertions if a.mask and not a.verdict]
+        return self._benign_assertions
+
+    @property
+    def valid_assertions(self):
+        if not self._valid_assertions:
+            self._valid_assertions = [a for a in self.assertions if a.mask]
+        return self._valid_assertions
+
+    @property
+    def filenames(self):
+        warnings.warn('This property is deprecated and will be removed in the next major version. '
+                      'Please use "Metadata().filenames" in the future.')
+        return []
```

Comment:
So, we don't have a way to do this correctly here anymore (meaning returning all the filenames) and returning an empty list has been the actual api behavior for sometime in the previous implementation. Since this does not break other existing code, I kept returning the empty list and I am warning the user it should use the `Metadata` resource instead.